discussion/src/main/java/org/openedx/discussion/presentation/comments/DiscussionCommentsViewModel.kt

```diff
@@ -66,7 +66,10 @@ class DiscussionCommentsViewModel(
 
     private val comments = mutableListOf<DiscussionComment>()
     private var page = 1
-    private var isLoading = false
+    private val _isLoading = MutableLiveData<Boolean>()
+    val isLoading: LiveData<Boolean>
+        get() = _isLoading
+    //private var isLoading = false
```

Comment:
Hi Mahesh, we should avoid removing or commenting out the original `isLoading` variable. It ensures that the next page isn’t requested while the current one is still loading. If you feel the name `isLoading` is ambiguous, you can go ahead and rename it.

For the loader of the new comment, I’d recommend creating a separate `MutableStateFlow` so both states remain independent. The same approach should also be applied in `DiscussionResponsesViewModel.kt`.

Why `MutableFlowState` over `LiveData`? Please [read](https://github.com/edx/edx-mobile-marketplace-android/blob/fix/navigation/docs/0001-strategy-for-data-streams.rst) this for it. 😉